hw03-Reflection/src/main/java/ru/otus/MyJUnit.java

```diff
@@ -0,0 +1,123 @@
+package ru.otus;
+
+import org.junit.jupiter.api.DisplayName;
+import ru.otus.annotations.Test;
+import ru.otus.annotations.After;
+import ru.otus.annotations.Before;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyJUnit {
+
+    private int failedTests = 0;
+    private int succedTests = 0;
+
+    public static void main(String[] args) {
+        MyJUnit myJUnit = new MyJUnit();
+        myJUnit.run("ru.otus.DIYArrayListTest");
+    }
+
+    public MyJUnit() {
+    }
+
+    public void run(String testClassName) {
+        Class<?> clazz;
+        try {
+            clazz = Class.forName(testClassName);
+        } catch (java.lang.Exception e) {
+            System.out.println("Тестовый класс " + testClassName + " не найден.");
+            return;
+        }
+
+        cleanStatistics();
+
+        Constructor<?>[] declaredConstructors = clazz.getDeclaredConstructors();
+        final List<Object> params = new ArrayList<Object>();
+
+        List<Method> beforeMethods = getMethodsAnnotatedWith(clazz, Before.class);
+        List<Method> testMethods = getMethodsAnnotatedWith(clazz, Test.class);
+        List<Method> afterMethods = getMethodsAnnotatedWith(clazz, After.class);
+
+        // Run each test
+        testMethods.stream().forEach(
+                testMethod -> {
+                    Object obj;
+                    // Для каждого теста новый экземпляр
+                    try {
+                        obj = declaredConstructors[0].newInstance();
```

Comment:
Можно было сам конструктор выше получить.

```suggestion
                        obj = constructor.newInstance();
```

Reply:
Переделал.